### PR TITLE
Slightly alter Keras dummy loss

### PR DESCRIPTION
**Pull request by Rocketknight1**
The existing `dummy_loss` returns a scalar, but this is incorrect - Keras expects loss functions to return a vector of per-sample losses. This only causes issues when the user passes `sample_weight` to `fit()` - I'll probably adjust our tests to include that to make sure we don't regress on this in future.

### CODE REVIEW

**Review by gante — APPROVED** [review]
LGTM 👍

**Comment by Rocketknight1** [comment]
@gante Can you take one more look at the expanded tests?

Also, the failing test looks like flakiness with a too-low threshold to me, unrelated to this PR

**Review by sgugger — APPROVED** [review]
Thanks for the slightly altering fix!
